internal/system/agent/services.go

```
@@ -172,86 +362,90 @@ func getConfig(services []string, ctx context.Context) (ConfigRespMap, error) {
-						m.Metrics[service] = fmt.Sprintf(err.Error())
-						LoggingClient.Error(err.Error())
-					} else {
-						m.Metrics[service] = ProcessResponse(responseJSON)
-					}
+				Configuration.Clients[e.ServiceId] = clientInfo
+				params := types.EndpointParams{
+					ServiceKey:  e.ServiceId,
+					Path:        "/",
+					UseRegistry: usingRegistry,
+					Url:         Configuration.Clients[e.ServiceId].Url() + clients.ApiMetricsRoute,
+					Interval:    internal.ClientMonitorDefault,
 				}
+				// TODO: The following note is related to future work:
+				// TODO: With the current deployment strategy, GeneralClient's func init(params types.EndpointParams) {...} returns blank "url"...
+				// TODO: [Need a manifest-like functionality in future...]
+				// Add the service key to the map where the value is the respective GeneralClient
+				generalClients[e.ServiceId] = general.NewGeneralClient(params, startup.Endpoint{RegistryClient: &registryClient})
 			}
-		} else {
-			// Service is known to SMA, so no need to ask the Registry for a ServiceEndpoint associated with `service`
-			// Simply use one of the ready-made list of clients.
-			LoggingClient.Info(fmt.Sprintf("service %s is known to SMA as being in the ready-made list of clients", service))
+		}
+	} else {
+		// Service is known to SMA, so no need to ask the Registry for a ServiceEndpoint associated with `service`
+		// Simply use one of the ready-made list of clients.
+		LoggingClient.Info(fmt.Sprintf("service %s is known to SMA as being in the ready-made list of clients", service))
 
-			responseJSON, err := generalClients[service].FetchMetrics(ctx)
-			if err != nil {
-				m.Metrics[service] = fmt.Sprintf(err.Error())
-				LoggingClient.Error(err.Error())
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
+		responseJSON, err := generalClients[service].FetchMetrics(ctx)
+		if err != nil {
+			//m.Metrics[service] = fmt.Sprintf(err.Error())
+			LoggingClient.Error(err.Error())
```

shouldn't we also return this error as well as log it?